javascriptv3/example_code/cross-services/transcription-app/src/recorder.ts

```diff
@@ -30,33 +35,30 @@ function handlerFunction(stream) {
       recordedAudio.src = URL.createObjectURL(blob);
       recordedAudio.controls = true;
       recordedAudio.autoplay = true;
+      // Take username from 'index.ts'.
+      var username = index.username;
+      upload(blob, username);
       sendData(blob);
-
-      const cognitoidentityserviceprovider = new AWS.CognitoIdentityServiceProvider(
-        { region: "eu-west-1" }
-      );
-      const userParams = {
-        AccessToken: getAccessToken(),
-      };
-      cognitoidentityserviceprovider.getUser(userParams, function (err, data) {
-        if (err) console.log(err, err.stack);
-        else console.log(data.Username);
-        var username = data.Username;
-        console.log("the username", username);
-        upload(blob, username);
-      });
-      // Upload recording to Amazon S3 bucket
-      alert("Refresh page in ~1 min to view your transcription.");
+      sendBlob(blob);
     }
   };
 }
+const sendBlob = async (blob) => {
+  try {
+    upload(blob, username);
+  } catch (err) {
+    console.log("Error", err);
+  }
+  // Upload recording to Amazon S3 bucket
```

Comment:
I suggest, "Upload the recording to the Amazon S3 bucket." (add the word "the" in two places plus the period at the end of the setnence.)